modules/configuring-the-lora-training-job.adoc

```diff
@@ -0,0 +1,249 @@
++
+[source,subs="+quotes"]
+----
+$ oc login __<openshift_cluster_url>__ -u __<username>__ -p __<password>__
+----
+
+. Configure a LoRA training job, as follows:
+.. Create a YAML file named `lora-config.yaml`.
+.. Add the following `ConfigMap` object definition:
++
+[source]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: lora-config
+  namespace: kfto
+data:
+  lora_config.json: |
+    {
+      "model_name_or_path": "bigscience/bloom-560m",
+      "training_data_path": "/data/input/twitter_complaints.json",
+      "output_dir": "/data/output/tuning/bloom-twitter",
+      "num_train_epochs": 10.0,
+      "per_device_train_batch_size": 4,
+      "gradient_accumulation_steps": 4,
+      "learning_rate": 1e-05,
+      "response_template": "\n### Label:",
+      "dataset_text_field": "output",
+      "use_flash_attn": false,
+      "lora_r": 8,
+      "lora_alpha": 8,
+      "lora_dropout": 0.1,
+      "bias": "none",
+      "target_modules": ["q_proj", "v_proj"]
+    }
+
+----
+.. Edit the training job metadata as shown in the following table.
++
+.Training job metadata
+[cols="16,84"]
+|===
+|Parameter | Value
+
+|`name`
+|Name of the training job
+
+|`namespace`
+|Name of your project
+|===
+
+.. Edit the parameters of the training job as shown in the following table.
++
+.Training job parameters
+[cols="30,70"]
+|===
+|Parameter | Value
+
+|`model_name_or_path`
+|Name of the pre-trained model
+
+|`training_data_path`
+|Path to the training data that you set in the `training_data.yaml` config map
+
+|`output_dir`
+|Output directory for the model
+
+|`num_train_epochs`
+|Number of epochs for training; in this example, the training job is set to run 10 times
+
+|`per_device_train_batch_size`
+|Batch size, the number of data set examples to process together; in this example, the training job processes 4 examples at a time
+
+|`gradient_accumulation_steps`
+|Number of gradient accumulation steps
+
+|`learning_rate`
+|Learning rate for the training
+
+|`response_template`
+|Template formatting for the response
+
+|`dataset_text_field`
+|Dataset field for training output, as set in the `training_data.yaml` config map
+
+|`use_flash_attn`
+|Whether to use flash attention
+
+|`lora_r`
+|Rank of the low-rank decomposition
+
+|`lora_alpha`
+|Scale the low-rank matrices to control their influence on the model's adaptations
+
+|`lora_dropout`
+|Dropout rate applied to the LoRA layers, a regularization technique to prevent overfitting
+
+|`bias`
+|Whether to adapt bias terms in the model; setting the bias to “none” indicates that no bias terms will be adapted
+
+|`target_modules`
+|Names of the modules to apply LoRA to; includes ‘q_proj” and ‘v_proj’ by default
+
+|===
+.. Save your changes in the `lora-config.yaml` file.
+.. Apply the configuration to create the `lora-config` object:
++
+[source]
+----
+$ oc apply -f lora-config.yaml
+----
+
+. Create the training data, as follows:
```

Comment:
The training data below should be treated as a very simple example, not suited for production use.
For any real case users will provide training data most likely by using persistent volumes. It may be good to mention it here, as current wording looks to me like preferred approach.

Reply:
@sutaakar Added a note as requested.
Should we also provide an example of how to provide the training data by using persistent volumes?

Reply:
@VanillaSpoon WDYT re PV example?

Reply:
If additional info is needed, we can provide in a separate PR later.